dmrunner/runner.py

```diff
@@ -103,6 +103,15 @@ def __init__(
         with open(self._settings_path) as settings_file:
             self.settings: dict = yaml.safe_load(settings_file.read())
 
+        # load environment variables from settings file,
+        # taking care not to override existing envvars
+        if self.settings.get("environment"):
+            for env in self.settings["environment"]:
+                key, value = env.split("=", 1)
```

Comment:
This is necessary because we're storing the environment values as `key=value`. Can we simplify this? Would a dictionary be easier to process than a list of `key=value` strings?

Reply:
Ah, yes, good point. I was trying to copy the style of envvars in compose files, but using a dict is more obvious. I've made the change.